packages/core/src/external/cda/remove-b64.ts

```diff
@@ -37,18 +47,39 @@ export function removeBase64PdfEntries(payloadRaw: string): {
         if (comp.section.entry) {
           //eslint-disable-next-line @typescript-eslint/no-explicit-any
           comp.section.entry = toArray(comp.section.entry).filter((entry: any) => {
-            const mediaType = entry.act?.text?.["_mediaType"]?.trim().toLowerCase();
-            if (
-              (BINARY_MIME_TYPES.includes(mediaType) || mediaType == undefined) &&
-              (entry.act?.text?.["_representation"]?.trim().toLowerCase() ===
-                b64Representation.toLowerCase() ||
-                entry.organizer?.component?.observationMedia?.value?.["_representation"]
-                  ?.trim()
-                  .toLowerCase() === b64Representation.toLowerCase())
-            ) {
-              isRemovedEntries = true;
-              b64Attachments.push(entry);
-              return false;
+            if (isConcernActEntry(entry)) {
+              const mediaType = entry.act?.text?.["_mediaType"]?.trim().toLowerCase();
+              if (
+                (mediaType == undefined || BINARY_MIME_TYPES.includes(mediaType)) &&
+                isB64Representation(entry.act?.text?.["_representation"])
+              ) {
+                isRemovedEntries = true;
+                b64Attachments.acts.push(entry);
```

Comment:
per my comment above: you already use `entry.act?` above, so you could do something like:
```
const act = entry.act;
if (!act) return true; (THIS IS PSEUDO CODE)
const mediaType = act.text....
....
b64Attachments.acts.push(act);
....
```

Essentially you have your hands on the act here, so not reason to push the entry, only to grab the act *again* in the process method.

Reply:
yup yup yup